features/admin.connections.v1/components/edit/settings/custom-authenticator-settings.tsx

```diff
@@ -0,0 +1,488 @@
+ */
+
+import Button from "@oxygen-ui/react/Button";
+import ActionEndpointConfigForm from "@wso2is/admin.actions.v1/components/action-endpoint-config-form";
+import { AuthenticationType, EndpointConfigFormPropertyInterface } from "@wso2is/admin.actions.v1/models/actions";
+import { validateActionEndpointFields } from "@wso2is/admin.actions.v1/util/form-field-util";
+import {
+    FederatedAuthenticatorInterface,
+    FederatedAuthenticatorListItemInterface
+} from "@wso2is/admin.identity-providers.v1/models";
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { FinalForm, FormRenderProps, FormSpy } from "@wso2is/form";
+import { EmphasizedSegment } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { getLocalAuthenticator } from "../../../api/authenticators";
+import {
+    getFederatedAuthenticatorDetails,
+    updateCustomAuthentication,
+    updateFederatedAuthenticator
+} from "../../../api/connections";
+import {
+    AuthenticationPropertiesInterface,
+    ConnectionInterface,
+    CustomAuthConnectionInterface,
+    EndpointAuthenticationType,
+    EndpointAuthenticationUpdateInterface
+} from "../../../models/connection";
+import { handleConnectionUpdateError } from "../../../utils/connection-utils";
+import "./custom-authenticator-settings.scss";
+
+/**
+ * Proptypes for the Custom Local Authenticator edit page component.
+ */
+export interface CustomAuthenticatorSettingsPagePropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Is the authenticator a custom local authenticator.
+     */
+    isCustomLocalAuthenticator: boolean;
+    /**
+     * Is the data still loading.
+     */
+    isLoading?: boolean;
+    /**
+     * Connection details.
+     */
+    connector: CustomAuthConnectionInterface | ConnectionInterface;
+    /**
+     * Callback to update the connector details.
+     */
+    onUpdate: (id: string, tabName?: string) => void;
+}
+
+/**
+ * Custom authenticator settings page.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const CustomAuthenticatorSettings: FunctionComponent<CustomAuthenticatorSettingsPagePropsInterface> = ({
+    isCustomLocalAuthenticator,
+    isLoading,
+    connector,
+    onUpdate,
+    ["data-componentid"]: componentId = "custom-authenticator-settings-page"
+}: CustomAuthenticatorSettingsPagePropsInterface): ReactElement => {
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    const [ authenticatorEndpoint, setAuthenticatorEndpoint ] = useState<EndpointConfigFormPropertyInterface>(null);
```

Comment:
If possible, let's move the state, effect functions and other methods to a dedicated component hook to keep presentation layer from the logic. It will reduce the file length to ~200 lines, improve readability and provide better separation of concerns. 

https://github.com/wso2/identity-apps/blob/master/docs/write-code/CODE_QUALITY.md#use-component-hooks-to-separate-the-component-logic-from-the-presentation-layer